include/boost/geometry/policies/robustness/get_rescale_policy.hpp

```diff
@@ -99,10 +105,30 @@ static inline void init_rescale_policy(Geometry1 const& geometry1,
         RobustPoint& min_robust_point,
         Factor& factor)
 {
-    // Get bounding boxes
-    model::box<Point> env = geometry::return_envelope<model::box<Point> >(geometry1);
-    model::box<Point> env2 = geometry::return_envelope<model::box<Point> >(geometry2);
-    geometry::expand(env, env2);
+    // Get bounding boxes (when at least one of the geometries is not empty)
+    model::box<Point> env;
+    if (geometry::num_points(geometry1) == 0
+        && geometry::num_points(geometry2) == 0)
+    {
+        return;
+    }
+    else if (geometry::num_points(geometry1) == 0)
+    {
+        geometry::envelope(geometry2, env);
+    }
+    else if (geometry::num_points(geometry2) == 0)
+    {
+        geometry::envelope(geometry1, env);
+    }
+    else
+    {
+        geometry::envelope(geometry1, env);
+        model::box<Point> env2 = geometry::return_envelope
+            <
+                model::box<Point>
+            >(geometry2);
+        geometry::expand(env, env2);
```

Comment:
Will in this else-branch the optimal envelope be generated in the spherical nor geographical CS? AFAIR the envelope should be calculated at once for all of the involved geometries.